FirebaseDatabase/Tests/Integration/FIRDatabaseQueryTests.m

```diff
@@ -2363,6 +2806,45 @@ - (void)testStartAtAndLimit1AndRemoveFirstChild {
   XCTAssertTrue([val isEqualToNumber:@2], @"Expected next element in window");
 }
 
+- (void)testStartAfterAndLimit1AndRemoveFirstChild {
+  FIRDatabaseReference* ref = [FTestHelpers getRandomNode];
+
+  __block NSNumber* val = nil;
+  FIRDatabaseQuery* query =
+      [[[ref queryOrderedByPriority] queryStartingAfterValue:nil
+                                                    childKey:@"a"] queryLimitedToFirst:1];
+  [query observeEventType:FIRDataEventTypeChildAdded
+                withBlock:^(FIRDataSnapshot* snapshot) {
+                  id newVal = [snapshot value];
+                  if (newVal != nil) {
+                    val = [snapshot value];
+                  }
+                }];
+
+  __block BOOL ready = NO;
+  [ref setValue:@{@"a" : @1, @"b" : @2}
+      withCompletionBlock:^(NSError* err, FIRDatabaseReference* ref) {
+        ready = YES;
+      }];
+
+  [self waitUntil:^BOOL {
+    return ready;
+  }];
+
+  XCTAssertTrue([val isEqualToNumber:@2], @"Expected first element in window");
+
+  ready = NO;
+  [[ref child:@"a"] removeValueWithCompletionBlock:^(NSError* err, FIRDatabaseReference* ref) {
+    ready = YES;
+  }];
+
+  [self waitUntil:^BOOL {
+    return ready;
+  }];
+
+  XCTAssertTrue([val isEqualToNumber:@2], @"Expected next element in window");
```

Comment:
Does `val` get updated when the second snapshot fires? I would expect that the observer doesn't get invoked.

Reply:
It does not. I ran this test with a break point and the observer only fired once with `@2`.

Reply:
Seems like we should remove this line then. It doesn't really expand on our test coverage.